api/app/tests/Unit/Controller/ClientControllerTest.php

```diff
@@ -214,11 +233,23 @@ public function testfindByIdAuth()
         $this->assertEndpointNeedsAuth('GET', $url);
     }
 
-    public function testfindByIdAcl()
+    public function testfindByIdAclNotAllowed()
+    {
+        $url = '/client/'.self::$primaryAccountClient->getId();
+
+        $this->assertEndpointNotAllowedFor('GET', $url, self::$tokenDeputy);
+    }
+
+    public function testfindByIdAclAllowed()
     {
-        $url2 = '/client/'.self::$client2->getId();
+        $url = '/client/'.self::$primaryAccountClient->getId();
+        $url2 = '/client/'.self::$nonPrimaryAccountClient->getId();
+
+        $this->assertEndpointAllowedFor('GET', $url, self::$tokenMultiClientPrimaryDeputy);
+        $this->assertEndpointAllowedFor('GET', $url2, self::$tokenMultiClientPrimaryDeputy);
 
-        $this->assertEndpointNotAllowedFor('GET', $url2, self::$tokenDeputy);
+        $this->assertEndpointAllowedFor('GET', $url, self::$tokenMultiClientNonPrimaryDeputy);
+        $this->assertEndpointAllowedFor('GET', $url, self::$tokenMultiClientNonPrimaryDeputy);
     }
```

Comment:
Add test to check a co-deputy not associated with the client can't access the client

Reply:
Also add a test for prof/pas (pa's accessing other pa clients, pa's accessing lay clients)